pkg/apis/extensions/types.go

```diff
@@ -296,6 +300,17 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int `json:"updatedReplicas,omitempty"`
+
+	// Stuck indicates if the current deployment has failed.
+	Stuck bool `json:"stuck,omitempty"`
```

Comment:
Rather than a boolean, we need to express failed as a condition (i.e. using `ConditionStatus`).